addons/stock/models/res_config_settings.py

```diff
@@ -53,6 +53,18 @@ class ResConfigSettings(models.TransientModel):
         "Separator", config_parameter='stock.barcode_separator',
         help="Character(s) used to separate data contained within an aggregate barcode (i.e. a barcode containing multiple barcode encodings)")
     module_stock_fleet = fields.Boolean("Dispatch Management System")
+    replenish_on_order = fields.Boolean("Replenish on Order (MTO)", config_parameter='replenish_on_order_settings', compute='_compute_replenish_on_order', inverse='_inverse_replenish_on_order')
+
+    @api.depends('replenish_on_order')
```

Comment:
A compute don't need to have the field that he computes in the depends but instead the fields that will retrigger the computation